docs/source/metrics.rst

```diff
@@ -196,12 +196,71 @@ Metric API
 .. autoclass:: pytorch_lightning.metrics.Metric
     :noindex:
 
-*************
-Class metrics
-*************
+***************************
+Class vs Functional Metrics
+***************************
 
+The functional metrics follow the simple paradigm input in, output out. This means, they don't provide any advanced mechanisms for syncing across DDP nodes or aggregation over batches. They simply compute the metric value based on the given inputs.
+
+Also the integration within other parts of PyTorch Lightning will never be as tight as with the class-based interface.
```

Comment:
I believe you are right. So the thing is that I didn't write this part - this is just taken from the current documentation (it's under Functional metrics right now). I think I rather leave rewording of this part to another PR :)

Also, for this suggestion: I think `input in, output out` is more descriptive than `input and output`, which was probably the intention of the author. Will add the comma tho :)

Reply:
@rohitgr7 it is true that metrics can be aggregated using `.log` but in some cases they will be aggregated wrongly.

Reply:
@SkafteNicki can you give an example?

Reply:
General problem:
Assuming that your number of samples `N` are not divisible by your batch size `B`, meaning that your last batch will be smaller than all the others. Since `.log` give uniform weight to all logged values when taking the average, the aggregated value will be biased towards the last batch.
A specific metric:
RMSE is one of the best examples, as it cannot be computed as a simple average over values calculated on individual batches (since `\sqrt(x+y)!=\sqrt(x)+\sqrt(y)`). To get this right, values inside the square root need to be aggregated first and the take the square root (which is what the metrics package does)

Reply:
oh nice!!